### PR TITLE
Handle riemann disconnects gracefully

### PR DESCRIPTION
**Pull request by mahendra**
@srikiraju please take a look

### CODE REVIEW

File: src/diamond/handler/riemann.py [hunk 1]
```diff
@@ -88,9 +91,13 @@ def process(self, metric):
         try:
             self.client.send_event(event)
         except Exception, e:
-            self.log.error(
+            self.log.exception(
                 "RiemannHandler: Error sending event to Riemann: %s", e)
 
+            # Re-open the transport. The riemann server may have gone away
+            self._close()
+            self._connect()
```
Comment (srikiraju) [inline src/diamond/handler/riemann.py:99]: Do you think it makes more sense to do this only on specific types of exceptions?

**Comment by srikiraju** [comment]
How are you going to package this? Are you going to upload to our artifactory?

Additionally, I noticed we use `diamond==4.0.809` but there seems to be no corresponding tag in Affirm fork or upstream...or on pypi...

File: src/diamond/handler/riemann.py (hunk 1 above)
Comment (mahendra) [reply src/diamond/handler/riemann.py:99]: I want to deploy this and see the type of exceptions that come up. We can handle the conditions in the future.

**Comment by mahendra** [comment]
@srikiraju https://github.com/Affirm/salt/blob/develop/srv/pillar/metrics.sls#L129 
https://github.com/Affirm/salt/blob/develop/srv/salt/metrics/diamond/install/pip.sls#L28

I pip install from github.

Once we merge to our branch, I will also send a PR upstream.

**Review by srikiraju — APPROVED** [review]

**Pull request merged**